atd-vze/src/views/Crashes/CrashChangeLog.js

```diff
@@ -1,243 +1,177 @@
-      content = (
-        <>
-          <h4>Record History</h4>
-
-          <Table responsive>
-            <thead>
+          <tbody className="text-monospace">
+            {selectedChange.diffs.map(diff => (
               <tr>
-                <td>Date Edited</td>
-                <td>Updated by</td>
-                <td></td>
+                <td className="text-monospace">{diff.field}</td>
+                {!isNewRecordEvent(selectedChange) && (
+                  <td className="text-monospace">{String(diff.old)}</td>
+                )}
+                <td className="text-monospace">{String(diff.new)}</td>
               </tr>
-            </thead>
-            <tbody>
-              {this.props.data.atd_txdot_change_log.map(record => (
-                <tr key={`changelog-${record.change_log_id}`}>
-                  <td>
-                    <Badge color="warning">
-                      {formatDateTimeString(record.update_timestamp)}
-                    </Badge>
-                  </td>
-                  <td>
-                    <Badge color="danger">
-                      {String(record.updated_by || "Unavailable")}
-                    </Badge>
-                  </td>
-                  <td>
-                    <Button
-                      color="primary"
-                      size="sm"
-                      onClick={() => this.compare(record)}
-                    >
-                      Compare
-                    </Button>
-                  </td>
-                </tr>
-              ))}
-            </tbody>
-          </Table>
-        </>
-      );
-    }
-
-    // Render Box
-    return (
-      <div className="animated fadeIn">
-        {modal}
-        <Row>
-          <Col>
-            <Card>
-              <CardHeader>
-                <i className="fa fa-history" /> Change Log
-              </CardHeader>
-              <CardBody>{content}</CardBody>
-            </Card>
-          </Col>
-        </Row>
-      </div>
-    );
+            ))}
+          </tbody>
+        </Table>
+      </ModalBody>
+      <ModalFooter>
+        <Button color="secondary" onClick={() => setSelectedChange(null)}>
+          Close
+        </Button>
+      </ModalFooter>
+    </Modal>
+  );
+};
+
+/**
+ * The primary UI component which renders the change log with clickable rows
+ */
+export default function CrashChangeLog({ data }) {
+  const [selectedChange, setSelectedChange] = useState(null);
+
+  const changes = useChangeLogData(data);
+  if (changes.length === 0) {
+    return <p>No change history found</p>;
   }
-}
 
-export default CrashChangeLog;
+  return (
+    <Card>
+      <CardHeader>Record history</CardHeader>
+      <CardBody>
+        <Table responsive striped hover>
+          <thead>
+            <th>Record type</th>
+            <th>Event</th>
+            <th>Affected fields</th>
+            <th>Edited by</th>
+            <th>Date</th>
+          </thead>
+          <tbody className="text-monospace">
```

Comment:
how do y'all feel about the monospace font? i don't love it, but i like it better than using the plain old san-serif, especially for the purposes of comparing data changes like `0` vs `O`

Reply:
I like the idea of using the monospace font, but I thought it looked too big -- I tried reducing the size to 90% and I thought it looked better, but when I went to check my screenshots, I couldnt tell a difference anymore
![Uploading Screenshot 2024-06-06 at 11.37.59 AM.png…]()

Reply:
<img width="320" alt="Screenshot 2024-06-06 at 11 38 05 AM" src="https://github.com/cityofaustin/atd-vz-data/assets/12474808/abf357fd-4511-434e-b5da-81b7344f6753">
this is full size. so yeah im just being nitpicky

Reply:
nitpicks are welcome 🙏 i'll try this out

Reply:
i opted to leave the font sizing as-is, i guess for accessibility? it will be good to to get feedback from users on this.

Reply:
i like the monospace, i think the font size is fine as is

Reply:
i'm leaning towards trying it out too. especially if there is an opportunity for it to improve direct comparison of edit values since this is a table. honestly can't remember if that is how this is even going to work at any point.